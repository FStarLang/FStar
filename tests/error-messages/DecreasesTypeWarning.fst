module DecreasesTypeWarning

let rec f (x:nat)  () : string =
  if x = 0 then "" else f (x - 1) ()
and g (xs:list nat) () : string =
  match xs with
  | [] -> ""
  | x::xs -> f x () ^ g xs ()
