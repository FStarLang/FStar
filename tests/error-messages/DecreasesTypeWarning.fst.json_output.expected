{"msg":["In the decreases clause for this function, the SMT solver may not be able to\nprove that the types of\n  xs (bound in DecreasesTypeWarning.fst(5,7-5,9))\nand x (bound in DecreasesTypeWarning.fst(3,11-3,12))\nare equal.","The type of the first term is: Prims.list Prims.nat","The type of the second term is: Prims.nat","If the proof fails, try annotating these with the same type."],"level":"Warning","range":{"def":{"file_name":"DecreasesTypeWarning.fst","start_pos":{"line":5,"col":7},"end_pos":{"line":5,"col":9}},"use":{"file_name":"DecreasesTypeWarning.fst","start_pos":{"line":5,"col":7},"end_pos":{"line":5,"col":9}}},"number":290,"ctx":["While typechecking the top-level declaration `let rec f and g`"]}
{"msg":["In the decreases clause for this function, the SMT solver may not be able to\nprove that the types of\n  x (bound in DecreasesTypeWarning.fst(3,11-3,12))\nand xs (bound in DecreasesTypeWarning.fst(5,7-5,9))\nare equal.","The type of the first term is: Prims.nat","The type of the second term is: Prims.list Prims.nat","If the proof fails, try annotating these with the same type."],"level":"Warning","range":{"def":{"file_name":"DecreasesTypeWarning.fst","start_pos":{"line":3,"col":11},"end_pos":{"line":3,"col":12}},"use":{"file_name":"DecreasesTypeWarning.fst","start_pos":{"line":3,"col":11},"end_pos":{"line":3,"col":12}}},"number":290,"ctx":["While typechecking the top-level declaration `let rec f and g`"]}
